### PR TITLE
feat: new init command

### PR DESCRIPTION
**Pull request by PatrickMenoti**
- Now part of root command;
- Vulcan templates on a list;
- Auto-suggested name;

### CODE REVIEW

File: pkg/cmd/init/utils.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	"github.com/aziontech/azion-cli/pkg/contracts"
+	"github.com/aziontech/azion-cli/pkg/logger"
+	"github.com/aziontech/azion-cli/utils"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"go.uber.org/zap"
+)
+
+func yesNoFlagToResponse(info *InitInfo) bool {
```
Comment (maxwelbm) [inline pkg/cmd/init/utils.go:18]: This function does not seem to me to be a utils, it is only used once

File: pkg/cmd/init/utils.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	if info.YesOption {
+		return info.YesOption
+	}
+
+	return false
+}
+
+func (cmd *InitCmd) createTemplateAzion(info *InitInfo) error {
```
Comment (maxwelbm) [inline pkg/cmd/init/utils.go:26]: a template.go file would not be more suitable for this function

File: pkg/cmd/init/utils.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	err = cmd.Rename(dir+"/webdev/"+typeLang, azionDir)
+	if err != nil {
+		logger.Debug("Error while copying files to current project directory", zap.Error(err))
+		return utils.ErrorMovingFiles
+	}
+
+	return nil
+}
+
+func (cmd *InitCmd) createJsonFile(options *contracts.AzionApplicationOptions, info *InitInfo) error {
```
Comment (maxwelbm) [inline pkg/cmd/init/utils.go:108]: This would go into the template

File: pkg/cmd/init/utils.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+				logger.Debug("Error while trying to clean azion directory", zap.Error(err))
+				return false, err
+			}
+		}
+		return shouldFetchTemplates, nil
+	}
+	return true, nil
+}
+
+func askForInput(msg string, defaultIn string) (string, error) {
```
Comment (maxwelbm) [inline pkg/cmd/init/utils.go:149]: I think you can add a struct input as input to make this func reusable? 
```suggestion
func askForInput(msg string, defaultIn string, prompt *survey.Input) (string, error) {
```

File: pkg/cmd/init/tags.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+	var previousMinor int = 0
+	var previousPatch int = 0
+
+	err = tags.ForEach(func(t *plumbing.Reference) error {
+		tagCurrent := t.Name().String() // return this format "refs/tags/v0.10.0"
+		if !strings.Contains(tagCurrent, "dev") {
+			versionParts := strings.Split(tagCurrent, ".")
+
+			majorCurrent, _ := strconv.Atoi(versionParts[0])
```
Comment (maxwelbm) [inline pkg/cmd/init/tags.go:26]: This code contains flaw, leave it to me 😲

**Review by gabriel-azion — APPROVED** [review]

File: pkg/cmd/init/utils.go (hunk 1 above)
Comment (PatrickMenoti) [reply pkg/cmd/init/utils.go:18]: Where would it go then?

**Review by gabriel-azion — APPROVED** [review]

**Review by maxwelbm — APPROVED** [review]

**Comment by maxwelbm** [comment]
Lgtm 🔥

File: pkg/cmd/init/utils.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+	// Prompt the user for input
+	err := survey.AskOne(prompt, &userInput, survey.WithKeepFilter(true))
+	if err != nil {
+		return "", err
+	}
+	return userInput, nil
+}
+
+func (cmd *InitCmd) selectVulcanTemplates(info *InitInfo) error {
```
[the remark this answers is not in the document]
Comment (PatrickMenoti) [reply pkg/cmd/init/utils.go:63]: This method lists all Vulcan templates + nextjs for faststore and allows the user to pick one

**Pull request merged**